lib/core/utils/preload.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Validated the preload object
+ * @param {Object | boolean} preload configuration object or boolean passed via the options parameter to axe.run
+ * @return {boolean}
+ * @private
+ */
+function isPreloadValidObject(preload) {
+	return (
+		typeof preload === 'object' &&
+		preload.hasOwnProperty('assets') &&
+		Array.isArray(preload.assets) &&
+		preload.assets.length
+	);
+}
+
+/**
+ * Returns a boolean which decides if preload is configured
+ * @param {Object} options run configuration options (or defaults) passed via axe.run
+ * @return {boolean}
+ */
+function shouldPreload(options) {
```

Comment:
It looks like this entire function could be simplified into a few LOC:

```js
const shouldPreload = options => {
  return options && isValidPreloadObject(options.preload)
}
```

Reply:
Nah, keeping this function as preload can be configured in various ways & this function keeps it clear to check. No harm.

Eg:
preload: true, // true | false (default) | { assets: ['cssom'], timeout: 30000 (optional) }